### PR TITLE
Improve: tidying up Conf.ml

### PR DESCRIPTION
**Pull request by gpetiot**
Just some cleaning and an optimization

### CODE REVIEW

File: src/Conf.ml [hunk 1]
```diff
@@ -1766,9 +1760,11 @@ let rec collect_files ~segs ~ignores ~enables ~files =
         if Fpath.exists filename then filename :: enables else enables
       in
       let files =
-        let filename = Fpath.(dir / dot_ocp_indent) in
-        if Fpath.exists filename then `Ocp_indent filename :: files
-        else files
+        let f_1 = Fpath.(dir / dot_ocamlformat) in
+        let f_2 = Fpath.(dir / dot_ocp_indent) in
+        files
+        |> (fun k -> if Fpath.exists f_1 then `Ocamlformat f_1 :: k else k)
```
Comment (Julow) [inline src/Conf.ml]: I think ``let files = if Fpath.exists f_1 then `Ocamlformat f_1 :: files else files in`` would be a little more readable. That way `f_1` and `f_2` could be defined just before they are used.

File: src/Conf.ml [hunk 2]
```diff
@@ -1839,14 +1834,13 @@ type action =
...
-  | Some kind -> kind
-  | None -> (
-    match Filename.extension fname with
-    | ".ml" -> `Impl
-    | ".mli" -> `Intf
-    | ".mlt" -> `Use_file
-    | _ -> `Impl )
+  Option.value !kind
+    ~default:
+      ( match Filename.extension fname with
```
Comment (Julow) [inline src/Conf.ml]: This was computed only when needed so this is slightly less good.

File: src/Conf.ml [hunk 3]
```diff
@@ -1872,9 +1866,9 @@ let is_in_listing_file ~quiet ~listings ~filename =
       try
         In_channel.with_file (Fpath.to_string listing_file) ~f:(fun ch ->
             let lines =
-              In_channel.input_lines ch
-              |> List.mapi ~f:(fun i s -> (i + 1, String.strip s))
-              |> List.filter ~f:(fun (_, l) -> not (drop_line l))
+              List.filter_mapi (In_channel.input_lines ch) ~f:(fun i s ->
+                  let l = String.strip s in
+                  Option.some_if (not (drop_line l)) (i + 1, l))
```
Comment (Julow) [inline src/Conf.ml]: `Option.some_if` is only 6 characters shorter than an `if then Some ... else None` so I'm not sure it's worth the complexity. We are using it a lot in OCamlformat so it's OK.

File: src/Conf.ml [hunk 4]
```diff
@@ -1933,39 +1923,33 @@ let build_config ~file =
...
     List.for_all files ~f
   in
+  let quiet = conf.quiet in
   if no_ocamlformat_files && not enable_outside_detected_project then (
     ( if not conf.quiet then
-      let reason =
-        match project_root with
-        | Some root ->
+      let why =
+        Option.value_map project_root ~default:"no project root was found"
```
Comment (Julow) [inline src/Conf.ml]: I think this is adding a lot of complexity and is only a single line shorter

File: src/Conf.ml (hunk 1 above)
Comment (gpetiot) [reply src/Conf.ml]: done

File: src/Conf.ml (hunk 3 above)
Comment (gpetiot) [reply src/Conf.ml]: done

File: src/Conf.ml (hunk 4 above)
Comment (gpetiot) [reply src/Conf.ml]: done

File: src/Conf.ml (hunk 2 above)
Comment (gpetiot) [reply src/Conf.ml]: done

**Pull request merged**